containers/src/Data/Map/Strict/Internal.hs

```diff
@@ -805,7 +805,7 @@ alter = go
     go :: Ord k => (Maybe a -> Maybe a) -> k -> Map k a -> Map k a
     go f !k Tip = case f Nothing of
                Nothing -> Tip
-               Just x  -> singleton k x
+               Just x  -> x `seq` singleton k x
```

Comment:
Why is this necessary? `singleton` is strict in the value already: http://hackage.haskell.org/package/containers-0.6.4.1/docs/src/Data.Map.Strict.Internal.html#singleton